docs/node/interceptors.md

```diff
@@ -0,0 +1,113 @@
+    try {
+      return await next(input);
+    } finally {
+      console.log("Completed activity", {activityType: input.activityType});
+    }
+  }
+}
+```
+
+### Authorization
+
+```ts
+import {
+  defaultDataConverter,
+  Next,
+  WorkflowInboundCallsInterceptor,
+  WorkflowInput,
+} from "@temporalio/workflows";
+
+/**
+ * WARNING: This demo is meant as a simple auth example.
+ * Do not use this for actual authorization logic.
+ * Auth headers should be encrypted and credentials
+ * stored outside of the codebase
+ */
+export class DumbWorkflowAuthInterceptor
+  implements WorkflowInboundCallsInterceptor
+{
+  public async execute(
+    input: WorkflowInput,
+    next: Next<WorkflowInboundCallsInterceptor, "execute">
+  ): Promise<unknown> {
+    const authHeader = input.headers.get("auth");
+    const {user, password} = authHeader
+      ? await defaultDataConverter.fromPayload(authHeader)
+      : undefined;
+
+    if (!(user === "admin" && password === "admin")) {
+      throw new Error("Unauthorized");
+    }
+    return await next(input);
+  }
+}
+```
+
+To properly do authorization from Workflow code, the Workflow would need to access encryption keys and possibly authenticate against an external user database, that requires the Workflow to break isolation using [external dependencies](/docs/node/external-dependencies).
+
+## Interceptor registration
+
+### Activity and client interceptors registration
+
+- Activity interceptors are registered on Worker creation by passing an array of [`ActivityInboundCallsInterceptor` factory functions](https://nodejs.temporal.io/api/interfaces/worker.activityinboundcallsinterceptorfactory) via [WorkerOptions](https://nodejs.temporal.io/api/interfaces/worker.workeroptions#interceptors).
+
+- Client interceptors are registered on `WorkflowClient` construction by passing an array of [`WorkflowClientCallsInterceptor` factory functions](https://nodejs.temporal.io/api/interfaces/client.workflowclientcallsinterceptorfactory) via [WorkflowClientOptions](https://nodejs.temporal.io/api/interfaces/client.workflowclientoptions#interceptors).
+
+### Workflow interceptors registration
+
+Workflow interceptor registration is different than the other interceptors because they run in the Workflow isolate, to register workflow interceptors, export an `interceptors` variable from a file located in the `workflows` directory and provide the name of that file to the Worker on creation via [WorkerOptions](https://nodejs.temporal.io/api/interfaces/worker.workeroptions#interceptors).
+
+`src/workflows/my-interceptors.ts`
+
+```ts
+export const interceptors = {
+  outbound: [new LogActivitiesInterceptor()],
+  inbound: [],
+};
+```
+
+At the time of construction, the Workflow Context is already initialized for the current Workflow.
+Use [`Context.info`](https://nodejs.temporal.io/api/interfaces/workflow.workflowinfo) to add Workflow specific information in the interceptor.
```

Comment:
still not very clear how to use, probably want to have a complete example in a single file